src/main/java/io/jenkins/plugins/gitlabserverconfig/servers/GitLabServer.java

```diff
@@ -249,9 +249,15 @@ public FormValidation doTestConnection(@QueryParameter String serverUrl,
             LOGGER.info("Testing Connection..");
             String privateToken = getToken(serverUrl, credentialsId);
             if (privateToken.equals(UNKNOWN_TOKEN)) {
-                LOGGER.error("Cannot find private token");
-                return FormValidation
-                        .errorWithMarkup(Messages.GitLabServer_credentialsNotResolved(Util.escape(credentialsId)));
+                GitLabApi gitLabApi = new GitLabApi(serverUrl, "");
+                try {
+                    gitLabApi.getProjectApi().getProjects(1, 1);
```

Comment:
Why the params are two `1`?

Reply:
This is only bound to go wrong :laughing: Not all instances have kept project with id 1 around.

Reply:
@casz I told this already. This `getProjects` method tells GitLab Server to return (no. of projects, no. of pages). So whtat getProjects(1,1) tells GitLab is return the latest project in 1 page. If there is no project found it returns `empty list`. If the server is invalid, it raises error. There is no project id used here.

Reply:
Add comment on it will be better.

Reply:
Fixed in a359410.